django-stubs/db/models/fields/__init__.pyi

```diff
@@ -202,7 +204,9 @@ class Field(RegisterLookupMixin, Generic[_ST, _GT]):
     def get_db_prep_save(self, value: Any, connection: BaseDatabaseWrapper) -> Any: ...
     def get_internal_type(self) -> str: ...
     # TODO: plugin support
-    def formfield(self, form_class: Any | None = ..., choices_form_class: Any | None = ..., **kwargs: Any) -> Any: ...
+    def formfield(
+        self, form_class: type[_Field] | None = ..., choices_form_class: Any | None = ..., **kwargs: Any
+    ) -> _Field: ...
```

Comment:
```suggestion
    def formfield(
        self, form_class: type[Field] | None = ..., choices_form_class: Any | None = ..., **kwargs: Any
    ) -> Field: ...
```

I propose to type it like this, because you can change the type of the returned field in this method: https://github.com/django/django/blob/f9e9526800c921ae393ff58826daed51587b1727/django/db/models/fields/__init__.py#L1125

Reply:
Isn't this an `@overload`?

Reply:
But yeah, this method is kind of designed to be overridden. Most fields customize it

Reply:
@sobolevn Agreed, I'll update this.